### PR TITLE
Expose GEOS make valid options for geos 3.10+ builds to Repair Geometries algorithm and make_valid expression function

### PR DESCRIPTION
**Pull request by nyalldawson**
Allows use of the newer refined GEOS make valid methods in QGIS.

### CODE REVIEW

File: resources/function_help/json/make_valid [hunk 1]
```diff
@@ -6,7 +6,20 @@
...
+  },
+  {
+    "arg": "method",
+    "description": "repair algorithm. May be either 'structure' or 'linework'. The 'linework' option combines all rings into a set of noded lines and then extracts valid polygons from that linework. The 'structure' method first makes all rings valid and then merges shells and subtracts holes from shells to generate valid result. Assumes that holes and shells are correctly categorized.",
+    "default": "structure",
+    "optional": true
+  },
+  {
+    "arg": "keep_collapsed",
+    "description": "if set to true, then components will be kept that have collapsed into a lower dimensionality. For example, a ring collapsing to a line, or a line collapsing to a point.",
```
Comment (DelazJ) [inline resources/function_help/json/make_valid:18]: Should it be this?
```suggestion
    "description": "if set to true, then components that have collapsed into a lower dimensionality will be kept. For example, a ring collapsing to a line, or a line collapsing to a point.",
```

**Review by nirvn — APPROVED** [review]

**Comment by agiudiceandrea** [comment]
Please see https://github.com/qgis/QGIS/issues/50888

**Comment by nyalldawson** [comment]
@agiudiceandrea I've looked at that, but there's nothing I can see wrong in the code. I think the real solution is to get the mac builds updated.

**Comment by agiudiceandrea** [comment]
Maybe rising the minimum GEOS version to 3.10 is needed, at least for macOS builds.

**Comment by agiudiceandrea** [comment]
It is very strange to me that the user reports the error:
`Caught unhandled QgsException:  "The structured method to make geometries valid requires a QGIS build based on GEOS 3.10 or later"`
while the should not be executed at all
```
#if GEOS_VERSION_MAJOR==3 && GEOS_VERSION_MINOR<10
  if ( mMethod == Qgis::MakeValidMethod::Structure )
  {
    throw QgsProcessingException( "The structured method to make geometries valid requires a QGIS build based on GEOS 3.10 or later" );
  }
#endif
```

**Pull request merged**